pkg/types/baremetal/validation/platform.go

```diff
@@ -32,6 +34,30 @@ func validateIPNotinMachineCIDR(ip string, n *types.Networking) error {
 	return nil
 }
 
+func validateOSImageURI(uri string, osimage string) error {
+	// Check for valid URI and sha256 checksum part of the URL
+	if strings.HasPrefix(uri, "http://") || strings.HasPrefix(uri, "https://") {
```

Comment:
You could check parsedURL.Scheme, `HasPrefix` is not reliable, as a user could give you `HTTPS://`, `.Scheme` normalizes it: https://play.golang.org/p/eoZ2JDSc8ZU

Reply:
+1. Thanks for the suggestion. Will use .Scheme